### PR TITLE
feat: prisma asset endpoint

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/3505)

- [x] This change addresses the issue in full

## Description
This creates the `presigned-upload-metadata` endpoint which is the only asset endpoint that gets used. 

## How Can This Be Tested/Reviewed?
The endpoint is available to test on http://localhost:3101/api endpoint

**please note you will have to update your .env file to include the new setting**

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend_new/.env.template [hunk 1]
```diff
@@ -2,4 +2,5 @@ DATABASE_URL="postgres://<userNameHere>@localhost:5432/bloom_prisma"
 PORT=3101
 GOOGLE_API_EMAIL=
 GOOGLE_API_ID=
-GOOGLE_API_KEY=
\ No newline at end of file
+GOOGLE_API_KEY=
+CLOUDINARY_SECRET=
```
Comment (seanmalbert) [inline backend_new/.env.template:6]: please add new line to EOF

**Review by seanmalbert — CHANGES_REQUESTED** [review]
@YazeedLoonat , I believe for Doorway the fellows abstracted the cloudinary dependency as part of their work to setup Bloom to work with s3. Can you check that out and incorporate that here, if it makes sense? cc @ludtkemorgan

**Comment by ludtkemorgan** [comment]
@seanmalbert  they did abstract it out but it will take extra work to make it work in core. @YazeedLoonat would you want to investigate it? If it's a lot of extra work we might want to pull it out as a separate ticket

**Comment by YazeedLoonat** [comment]
Hey @seanmalbert @ludtkemorgan I think I'm a lil hesitant to set up AWS work when we don't have anything set up for using AWS. 

As it stands right now the AWS call is done through an endpoint we don't leverage in our current app mainly the `upload` endpoint under asset.controller.ts

If you are both okay with it I think my preference would be to spin up a ticket when tied to a move to AWS epic maybe

**Comment by YazeedLoonat** [comment]
> > Hey @seanmalbert @ludtkemorgan I think I'm a lil hesitant to set up AWS work when we don't have anything set up for using AWS.
> > As it stands right now the AWS call is done through an endpoint we don't leverage in our current app mainly the `upload` endpoint under asset.controller.ts
> > If you are both okay with it I think my preference would be to spin up a ticket when tied to a move to AWS epic maybe
> 
> @YazeedLoonat , @ludtkemorgan discussed this with me yesterday and what you're proposing sounds good.
> 
> For the ticket you spin up, is the 3rd party service abstraction work dependent on the AWS piece being ready? I ask because, ideally the abstraction was done in a way that allows for specifying either AWS s3 or Cloudinary as the asset management provider, so in this case you could bring in the abstraction with Cloudinary working first and not add in s3 until the AWS piece is ready.


@seanmalbert @ludtkemorgan 
I'm not sure if a github pr is the best place to track this discussion but:

The way the flow works in bloom right now is:
we get signed meta data from asset controller's `presigned-upload-metadata` endpoint, but the partner site's front end does the actual upload to cloudinary. Our current backend doesn't actually manage any part of that upload process which is my preference. And is similar to how civiform does asset uploads as well (at least from what I remember)

Doorway https://github.com/metrotranscom/doorway/blob/bdaf76e6b6436275dd93495ae3e101738b6e13e6/backend/core/src/assets/services/assets.service.ts#L29 has a dedicated upload endpoint that does the upload itself. Doorway also deactivated the cloudinary upload flow so we would need to do some work to get that back up and running. 

I think broadly my preference is to move back to the partner site doing the file upload and sending to the backend the outcome of that upload during a listing save. I think its kinda unnecessary for our backend to be an intermediate in this step. That is of course if we can do that I'm not sure with the AWS VPC in place. 

I think in terms of setting up the AWS flow would be bringing over that `upload` endpoint from doorway. But I think that we may want to make a decision on which flow ends up working/working the best 

hopefully that all makes sense

**Comment by ludtkemorgan** [comment]
@YazeedLoonat Those are all valid points about the flow. I think we should probably have this discussion on the new ticket you create. I had a discussion months back with Chris and the Google team on why everything was moved to the backend, I don't remember all of the reasons but I don't think it was solely because of the AWS setup

**Review by ludtkemorgan — APPROVED** [review]

File: backend_new/src/services/asset.service.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@nestjs/common';
+import { v2 as cloudinary } from 'cloudinary';
+import { CreatePresignedUploadMetadata } from '../dtos/assets/create-presigned-upload-meta.dto';
+import { CreatePresignedUploadMetadataResponse } from '../dtos/assets/create-presign-upload-meta-response.dto';
+
+/*
+  this is the service for unit types
+  it handles all the backend's business logic for reading/writing/deleting unit type data
+*/
```
[the remark this answers is not in the document]
Comment (YazeedLoonat) [reply backend_new/src/services/asset.service.ts]: yikes, got that updated ty

File: backend_new/src/services/asset.service.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+@Injectable()
+export class AssetService {
+  /*
+    this will get a set of unit types given the params passed in
+  */
```
[the remark this answers is not in the document]
Comment (YazeedLoonat) [reply backend_new/src/services/asset.service.ts]: same

**Comment by emilyjablonski** [comment]
@ludtkemorgan is this ready to merge?

**Comment by ludtkemorgan** [comment]
> @ludtkemorgan is this ready to merge?

This one had some outstanding questions from Sean. @YazeedLoonat do you think it is good to merge?

**Review by seanmalbert — APPROVED** [review]
@YazeedLoonat , please see my comment and address before merging.

**Pull request merged**